src/DevTo.Api/ListingsService.cs

```diff
@@ -0,0 +1,78 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Dynamic;
+using System.Linq;
+using System.Net.Http;
+using System.Reflection;
+using System.Threading.Tasks;
+using Forem.Api.Models;
+using Forem.Api.Models.Listing;
+using Newtonsoft.Json;
+
+namespace Forem.Api
+{
+	public class ListingsService : ApiService, IListingsApi
+	{
+		public ListingsService(Uri baseUri, HttpClient httpClient) : base(baseUri, httpClient) { }
+
+		public Task<IEnumerable<Listing>> GetListingsAsync(int page = 1, int perPage = 30)
+			=> GetAsync<IEnumerable<Listing>>("/api/listings", new Dictionary<string, object>
+				{
+					{ "page", page },
+					{ "per_page", perPage },
+				});
+
+		public Task<Listing> CreateListingAsync(string apiKey, ListingProperties properties, ListingAction action = ListingAction.Create)
+			=> PostAsync<Listing>("/api/listings", CreateRequestPayload(properties, action), apiKey);
+
+		public Task<IEnumerable<Listing>> GetListingsByCategoryAsync(ListingCategory category, int page = 1, int perPage = 30)
+			=> GetAsync<IEnumerable<Listing>>($"/api/listings/category/{category}", new Dictionary<string, object>
+				{
+					{ "page", page },
+					{ "per_page", perPage },
+				});
+
+		public Task<Listing> GetListingAsync(long id) => GetAsync<Listing>($"/api/listings/{id}");
+
+		public Task<CompleteListing> UpdateListingAsync(string apiKey, long id, ListingAction action, ListingProperties properties = null)
+			=> PutAsync<CompleteListing>($"/api/listings/{id}", CreateRequestPayload(properties, action), apiKey);
+
+		private dynamic CreateRequestPayload(ListingProperties properties, ListingAction action)
+		{
+			var listing = new ExpandoObject() as IDictionary<string, object>;
```

Comment:
Rather than doing an `ExpandoObject` here, directly creating a `Dictionary<string, object>` would be easier. Additionally you won't need the `dynamic` return type, instead return an `IDictionary<string, object>`.

This might also "fix" whatever weirdness is going on with compiling the code on AppVeyor: https://ci.appveyor.com/project/Turnerj/forem-dotnet/builds/33644472#L63

(That said, I cloned it locally and it compiled fine so I have no idea.)